benchmark/main.cpp

```diff
@@ -127,15 +127,21 @@ void benchmark(size_t colSize, int colCount, int threadCount, int iterations, bo
             clearCache();
         }
 
-        for (int j = 0; j < threadCount; j++) {
+        for (int j = 0; j < threadCount - 1; j++) {
             size_t endIndex = startIndex + partLength + (j < overhang ? 1 : 0);
             auto threadInstance = new thread(threadFunc<T>, ref(attributeVector), colCount, startIndex,
                     endIndex, j);
             threads.push_back(threadInstance);
             startIndex = endIndex;
         }
+
         threadFlag = true;
 
+        // run threadFunc on main thread
+        int j = threadCount - 1;
```

Comment:
this is not necessary if you use replace "j" in the line below with threadCount - 1

Reply:
I'd rather keep it for consistency with the call above